File: Python/Ego.py
coordinators = ("DAWID", "GRZESIEK", "MACIEK", "MARCIN", "MARIUSZ", "RAFAL")
t = int(input())
for _ in range(t):
    n = int(input())
    print(coordinators[(n - 1) // 7 % len(coordinators)])

File: Java/Ego.java
import java.util.Scanner;

public class Main
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        final String[] coordinators = {"DAWID", "GRZESIEK", "MACIEK", "MARCIN", "MARIUSZ", "RAFAL"};
        int t = input.nextInt();
        for (int i = 0; i < t; i++) 
        {
            int n = input.nextInt();
            System.out.println(coordinators[(n - 1) / 7 % coordinators.length]);
        }
        input.close();
    }
}